### PR TITLE
improvement(tls-certs): collect and keep TLS/SSL artifacts

### PR DESCRIPTION
**Pull request by dimakr**
Collect TLS/SSL artifacts (server, client certificates; CA certificate; etc.) and keep them after a test is finished, similarly to how logs are collected and published.
This will facilitate root cause analysis of SCT failures caused by certificate related issues (e.g. "Unknown Subject Alternative name in X.509 certificate" kind of errors, etc.).

Closes: https://github.com/scylladb/scylla-cluster-tests/issues/9133

### Testing
- [x] :green_circle: [pr-provision-test with enabled encryption, certs are collected and published to argus](https://argus.scylladb.com/tests/scylla-cluster-tests/a429edb6-f820-49d0-a68e-bb7383daa891)
- [x] :green_circle: [pr-provision-test with no encryption enabled, certs are not collected/published](https://argus.scylladb.com/tests/scylla-cluster-tests/0b5adecf-1ccc-4147-bb19-c54de7641ef1)

### PR pre-checks (self review)

- [ ] I added the relevant `backport` labels
- [ ] I didn't leave commented-out/debugging code

### Reminders

- Add New configuration option and document them (in `sdcm/sct_config.py`)
- Add unit tests to cover my changes (under `unit-test/` folder)
- Update the Readme/doc folder relevant to this change (if needed)

### CODE REVIEW

File: sdcm/logcollector.py [hunk 1]
```diff
@@ -1297,6 +1297,17 @@ def collect_logs(self, local_search_path: Optional[str] = None) -> list[str]:  #
...
 
+class CertsCollector(BaseSCTLogCollector):
+    log_entities = [
+        DirLog(name='ssl_conf/*', search_locally=True),
+    ]
+    cluster_log_type = 'ssl-conf'
+    cluster_dir_prefix = 'ssl-conf'
+
+    def collect_logs(self, local_search_path: Optional[str] = None) -> list[str]:
+        return super().collect_logs(local_search_path=local_search_path)
```
Comment (vponomaryov) [inline sdcm/logcollector.py:1308]: Why is it needed?

File: sdcm/tester.py [hunk 2]
```diff
@@ -2986,6 +2991,7 @@ def tearDown(self):
         time.sleep(1)  # Sleep is needed to let final event being saved into files
         self.save_email_data()
         self.argus_collect_gemini_results()
+        self.collect_certs()
```
Comment (vponomaryov) [inline sdcm/tester.py:2994]: Shouldn't it be called before the line `2988` (`self.clean_resources()`)?

File: sdcm/logcollector.py [hunk 3]
```diff
@@ -1297,6 +1297,17 @@ def collect_logs(self, local_search_path: Optional[str] = None) -> list[str]:  #
         return [s3_link]
 
 
+class CertsCollector(BaseSCTLogCollector):
+    log_entities = [
+        DirLog(name='ssl_conf/*', search_locally=True),
+    ]
+    cluster_log_type = 'ssl-conf'
+    cluster_dir_prefix = 'ssl-conf'
```
Comment (vponomaryov) [inline sdcm/logcollector.py:1305]: Typo? `ssl_conf`?

**Review by vponomaryov — CHANGES_REQUESTED** [review]

File: sdcm/logcollector.py (hunk 1 above)
Comment (dimakr) [reply sdcm/logcollector.py]: Removed. 
I was looking at/repeated another k8s related collector, but It is not needed indeed.

File: sdcm/tester.py (hunk 2 above)
Comment (dimakr) [reply sdcm/tester.py:2994]: Certs are collected from SCT runner, so cleaning cluster resources doesn't affect collecting certs.

Comment (fruch) [reply sdcm/tester.py:2994]: we might also consider collecting from nodes, to validate we didn't mess-up transferring them to the correct nodes ?

**Review by vponomaryov — DISMISSED** [review]
LGTM

File: sdcm/tester.py (hunk 2 above)
Comment (vponomaryov) [reply sdcm/tester.py:2994]: Gathering certs from nodes is also interesting, IMHO.
But we can do it later.

Comment (dimakr) [reply sdcm/tester.py:2994]: Added gathering certificates and cqlshrc from nodes (we had an a failure once with the error `configparser.DuplicateOptionError: While reading from '/home/ubuntu/.cassandra/cqlshrc' [line  8]: option 'usercert' in section 'ssl' already exists`, so collecting cqlshrc would help in investigating, if it occurs again)

**Comment by dimakr** [comment]
Failed unittest - `tests/test_base_version.py::TestBaseVersion::test_2024_2` - is not related to the change (it is probably a post-effect of promoting 2024.2 yesterday at the evening)

File: sdcm/tester.py (hunk 2 above)
Comment (vponomaryov) [reply sdcm/tester.py:2994]: So, probably now it should be moved before the step where DB nodes get cleaned up?

Comment (dimakr) [reply sdcm/tester.py:2994]: But maybe it is better to move the 'collect_certs` method before `clean_resources` one. If it confused once, will confuse again. And rename it to smth like collect_ssl_conf

**Review by vponomaryov — CHANGES_REQUESTED** [review]
Screenshot from 2024-11-20 19-27-53

Please, use DB node names for directories names like it is done in DB logs.

**Review by fruch — APPROVED** [review]
LGTM

even with the ip address, we don't need this that often

**Pull request merged**